packages/apps/dev-wallet/src/modules/wallet/wallet.service.ts

```diff
@@ -58,6 +63,21 @@ export function sign(
   return signedTx;
 }
 
+export function getRelevantKeys(
+  keySources: IKeySource[],
+  Tx: IUnsignedCommand,
+) {
+  return keySources.flatMap((keySource) => {
+    const { keys } = keySource;
+    const cmd: IPactCommand = JSON.parse(Tx.cmd);
+    const publicKeys = cmd.signers
+      .map((signer) => keys.find((key) => key.publicKey === signer.pubKey))
+      .filter((index) => index !== undefined) as Array<IKeyItem>;
```

Comment:
why do we need the cast here?

Reply:
TS is not smart enough to grasp the code filters out undefined items, the type it infers is  `Array<IKeyItem | undefined>` so I cast it explicitly to the correct type `Array<IKeyItem>`